handler/oauth2/flow_authorize_code_token_test.go

```diff
@@ -28,62 +28,80 @@ func TestAuthorizeCode_PopulateTokenEndpointResponse(t *testing.T) {
 		t.Run("strategy="+k, func(t *testing.T) {
 			store := storage.NewMemoryStore()
 
-			var h GenericCodeTokenEndpointHandler
-
 			testCases := []struct {
 				description string
 				areq        *fosite.AccessRequest
-				setup       func(t *testing.T, areq *fosite.AccessRequest, config *fosite.Config)
+				authreq     *fosite.AuthorizeRequest
+				setup       func(t *testing.T, areq *fosite.AccessRequest, authreq *fosite.AuthorizeRequest, config *fosite.Config)
 				check       func(t *testing.T, aresp *fosite.AccessResponse)
 				expectErr   error
 			}{
 				{
-					description: "should fail because not responsible",
+					description: "should fail because not responsible for handling the request",
 					areq: &fosite.AccessRequest{
-						GrantTypes: fosite.Arguments{"123"},
+						GrantTypes: fosite.Arguments{string(fosite.GrantTypeImplicit)},
+						Request: fosite.Request{
+							Client: &fosite.DefaultClient{
+								GrantTypes: fosite.Arguments{string(fosite.GrantTypeAuthorizationCode)},
+							},
+							Session:     &fosite.DefaultSession{},
+							RequestedAt: time.Now().UTC(),
+						},
 					},
 					expectErr: fosite.ErrUnknownRequest,
 				},
 				{
 					description: "should fail because authorization code cannot be retrieved",
 					areq: &fosite.AccessRequest{
-						GrantTypes: fosite.Arguments{"authorization_code"},
+						GrantTypes: fosite.Arguments{string(fosite.GrantTypeAuthorizationCode)},
 						Request: fosite.Request{
 							Form: url.Values{},
 							Client: &fosite.DefaultClient{
-								GrantTypes: fosite.Arguments{"authorization_code"},
+								GrantTypes: fosite.Arguments{string(fosite.GrantTypeAuthorizationCode)},
```

Comment:
Perhaps these changes should be undone, I think that it is intentional that they hard code the string in these tests to catch any typos in the var definition